### PR TITLE
Feature eapp 19 404 error page

### PR DESCRIPTION
**Pull request by animeshrick**
### 404 error page

> [JIRA Ticket:[ ECOMAPP - 1234 ](https://schoolbook.atlassian.net/browse/ECOMAPP-1234)](https://shoppixa.atlassian.net/browse/EAPP-19)

**What was the problem:**
404 Error Page -- Page not found with any anonymous URL

**How was this solved:**
Adding custom 404 page design

**Features:**
- Try it locally http://localhost:4000/puchi
- Showing an UI

**Services to be impacted:**
NONE

- `ecom-app`

**Example Code or Screenshot:**
404-page

**_For the PR Creator_**

- [x] Filled out entirely of PR Template.
- [x] Changes meet acceptance criteria defined in ticket.
- [ ] Unit Tests Passed. :tada:
- [x] Tested on DEV Env.
- [ ] Tested on QA Env.

### CODE REVIEW

File: src/components/home_page/categories.tsx [hunk 1]
```diff
@@ -23,12 +27,23 @@ const CategoriesList = () => {
...
         { id: 10, name: 'Toys & Gifts', icon: <ToysIcon/>},
+        { id: 11, name: 'Wine', icon: <Liquor /> },
+        { id: 12, name: 'Motor Accessories', icon: <DirectionsBike /> },
+        { id: 13, name: 'Demo', icon: <DirectionsBike /> },
+        { id: 14, name: 'Demo-1', icon: <DirectionsBike /> },
+        { id: 15, name: 'Demo-2', icon: <DirectionsBike /> },
+        { id: 16, name: 'Demo-3', icon: <DirectionsBike /> },
       ];
       
+      const listGroupStyle : CSS.Properties = {
```
Comment (KoushikMallik-developer) [inline src/components/home_page/categories.tsx:38]: Please keep all the codes in the particular way, CSS should be in a .css file and this style should be imported here.

File: src/index.css [hunk 2]
```diff
@@ -56,9 +57,31 @@ code {
...
 }
+
 @media (min-width: 769px) {
   .gradient-custom-2 {
     border-top-right-radius: .3rem;
     border-bottom-right-radius: .3rem;
   }
+}
+
+/* ::-webkit-scrollbar {
```
Comment (KoushikMallik-developer) [inline src/index.css:68]: Please remove any unwanted or commented code.

File: src/types/user_types/user.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+type User = {
```
Comment (KoushikMallik-developer) [inline src/types/user_types/user.ts:1]: Please add `delivery_address` field also.

File: src/components/404_error/page_not_found.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  MDBContainer,
+  MDBRow,
+  MDBCol,
+  MDBIcon,
+  MDBCardImage,
+  MDBBtn,
+} from "mdb-react-ui-kit";
+import errorImage from "../../components/404_error/404.png";
+
+const NotFoundPage = () => {
```
Comment (KoushikMallik-developer) [inline src/components/404_error/page_not_found.tsx:11]: please follow the naming convention. You can make it `Error404Page`.

File: src/components/404_error/404.png
Comment (KoushikMallik-developer) [inline src/components/404_error/404.png:1]: Please move this file to the particular assets folder.

**Review by KoushikMallik-developer — CHANGES_REQUESTED** [review]

File: src/types/user_types/user.ts (hunk 3 above)
Comment (animeshrick) [reply src/types/user_types/user.ts:1]: There is no use of User Type so be as it is

Comment (KoushikMallik-developer) [reply src/types/user_types/user.ts:1]: if there is no use of user type in this page then remove it from the PR.

**Pull request merged**